src/main/scala/chisel3/util/Decoupled.scala

```diff
@@ -215,18 +220,19 @@ class Queue[T <: Data](val gen: T,
     }
   }
 
-  val io = IO(new QueueIO(genType, entries))
-
+  val io = IO(new QueueIO(genType, entries, hasFlush))
   val ram = if (useSyncReadMem) SyncReadMem(entries, genType, SyncReadMem.WriteFirst) else Mem(entries, genType)
   val enq_ptr = Counter(entries)
   val deq_ptr = Counter(entries)
   val maybe_full = RegInit(false.B)
-
   val ptr_match = enq_ptr.value === deq_ptr.value
   val empty = ptr_match && !maybe_full
   val full = ptr_match && maybe_full
   val do_enq = WireDefault(io.enq.fire())
   val do_deq = WireDefault(io.deq.fire())
+  val flush = io.flush.getOrElse(false.B) 
+
+  // when flush is on, empty the queue
```

Comment:
This comment seems misplaced now

Reply:
you might also want to say below where you move the comment to the when(flush) clause:

```
// When flush is asserted, empty the queue.
// Semantically, any enqueues happen before the flush.
```